File: 0086-partition-list/0086-partition-list.cpp
class Solution {
public:
    ListNode* partition(ListNode* head, int x) {
        ListNode* lo = new ListNode(100);
        ListNode* hi = new ListNode(234);
        ListNode* tempLo = lo ;
        ListNode* tempHi = hi ;
        ListNode* temp = head ;
        while(temp!=NULL){
            if(temp->val<x){
                tempLo->next = temp;
                temp = temp->next;
                tempLo = tempLo->next;
            }
            else{//temp->val >=x
                tempHi->next = temp;
                temp = temp->next;
                tempHi = tempHi->next;
            }
        }
        tempLo->next = hi->next;//to connect the lower partition with the higher list 
        tempHi->next = NULL;// after all partition of low and high and after combination of low list with high in serires the temphi must have address of null to make the ll complete 
        return lo->next ;//this will return the complete list in order of the question asked 
    }
};

File: 0086-partition-list/0086-partition-list.java
class Solution {
    public ListNode partition(ListNode head, int x) {
        // creating two dummy nodes 
        ListNode d1 = new ListNode(-1);
        ListNode d2 = new ListNode(-1);
        ListNode t1 = d1;// for smaller part of linked list
        ListNode t2 = d2;// for larger part of linked list
        ListNode t = head;
        while(t!=null){
            if(t.val<x){
                t1.next = t;
                t1 = t1.next;
                t = t.next;
            }
            else{
                t2.next = t;
                t2 = t2.next;
                t = t.next;
            }
        }
        t1.next = d2.next;
        t2.next = null;
        return d1.next;
    }
}